pwa/src/components/sources/sourceForm.tsx

```diff
@@ -37,25 +25,98 @@ export const SourceForm: React.FC<SourceFormProps> = ({ sourceId }) => {
   const [_, setAlert] = React.useContext(AlertContext);
   const [__, setHeader] = React.useContext(HeaderContext);
 
-  React.useEffect(() => {
-    setHeader(
-      <>
-        Source <i>{source && source.name}</i>
-      </>,
-    );
-  }, [setHeader, source]);
+  const typeSelectOptions: ISelectValue[] = [
+    { label: "JSON", value: "json" },
+    { label: "SML", value: "xml" },
+    { label: "SOAP", value: "soap" },
+    { label: "FTP", value: "ftp" },
+    { label: "SFTP", value: "sftp" },
+  ];
+
+  const authSelectOptions: ISelectValue[] = [
+    { label: "API Key", value: "apikey" },
+    { label: "JWT", value: "jwt" },
+    { label: "Username and Password", value: "username-password" },
+  ];
+
+  const {
+    register,
+    control,
+    setValue,
+    getValues,
+    handleSubmit,
+    formState: { errors },
+  } = useForm();
 
   React.useEffect(() => {
+    setHeader("Source");
+
     handleSetDocumentation();
     sourceId && handleSetSource();
-  }, [sourceId, API]);
+  }, [sourceId, API, setHeader]);
+
+  const onSubmit = (data): void => {
+    data.type = data.type && data.type.value;
+    data.auth = data.auth && data.auth.value;
+
+    API.Source.createOrUpdate(data, sourceId)
+      .then(() => {
+        setAlert({ type: "success", message: `${sourceId ? "Updated" : "Created"} source` });
+        navigate("/sources");
+      })
+      .catch((err) => {
+        setAlert({ type: "danger", message: err.message });
+        throw new Error("Create or update source error: " + err);
+      })
+      .finally(() => {
+        setLoadingOverlay(false);
```

Comment:
You are setting the LoadingOverlay to false but you forgot the line of `setLoadingOverlay(true);` on line 64 so the spinner is never shown.

Reply:
Thanks!